File: mambo-protocol/src/main/scala/org/mambo/protocol/types/game/context/roleplay/HumanOptionGuild.scala





















// Generated on 05/03/2013 10:35:05
// Scala version by Blackrush
package org.mambo.protocol.types

import org.mambo.core.io._
import org.mambo.protocol.{Type => BaseType, _}
import org.mambo.protocol.enums._

trait HumanOptionGuild extends HumanOption {
  override val typeId = 409

  protected trait Type extends HumanOption.Type with Serializable {
    override val typeId = HumanOptionGuild.typeId

    val guildInformations: GuildInformations.TypeImpl

    override def serialize(buf: Buffer) {
      super.serialize(buf)

      guildInformations.serialize(buf)
    }

  }

  protected def create(guildInformations: GuildInformations.TypeImpl): Target = throw new IllegalAccessException

  override def deserialize(buf: Buffer) = {
    val guildInformations = GuildInformations.deserialize(buf)

    create(guildInformations)
  }

}

object HumanOptionGuild extends HumanOptionGuild {
  case class TypeImpl(guildInformations: GuildInformations.TypeImpl) extends Type
  override def create(guildInformations: GuildInformations.TypeImpl) = TypeImpl(guildInformations)

  def unapply(msg: TypeImpl) = TypeImpl.unapply(msg)
  def apply(guildInformations: GuildInformations.TypeImpl) = create(guildInformations)

  type Target = TypeImpl

}

File: mambo-protocol/src/main/java/org/mambo/protocol/types/HumanOptionGuild.java


// Generated on 05/08/2013 19:38:01
package org.mambo.protocol.types;

import java.util.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class HumanOptionGuild extends HumanOption {
    public static final short TYPE_ID = 409;
    
    @Override
    public short getTypeId() {
        return TYPE_ID;
    }
    
    public GuildInformations guildInformations;
    
    public HumanOptionGuild() { }
    
    public HumanOptionGuild(GuildInformations guildInformations) {
        this.guildInformations = guildInformations;
    }
    
    @Override
    public void serialize(Buffer buf) {
        super.serialize(buf);
        guildInformations.serialize(buf);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        super.deserialize(buf);
        guildInformations = new GuildInformations();
        guildInformations.deserialize(buf);
    }
    
}
